src/condor_negotiator.V6/matchmaker.cpp

```diff
@@ -3701,14 +3701,20 @@ Matchmaker::TransformSubmitterAd(classad::ClassAd &ad)
 		return true;
 	}
 
-	std::string authenticated_identity;
-	if (!ad.EvaluateAttrString(ATTR_AUTHENTICATED_IDENTITY, authenticated_identity)) {
+	std::string submitter_name;
+	if (!ad.EvaluateAttrString(ATTR_NAME, submitter_name)) {
 		return true;
 	}
+	auto last_at = submitter_name.find_last_of('@');
+	if (last_at == std::string::npos) {
+		return true;
+	}
+	auto accounting_domain = submitter_name.substr(last_at + 1);
+
 	std::vector<classad::ExprTree*> args;
 	args.reserve(2);
 	args.push_back(classad::Literal::MakeString("GROUP_PREFIX"));
-	args.push_back(ad.LookupExpr(ATTR_AUTHENTICATED_IDENTITY)->Copy());
+	args.push_back(classad::Literal::MakeString(accounting_domain));
 	std::unique_ptr<ExprTree> fnCall(classad::FunctionCall::MakeFunctionCall("userMap", args));
```

Comment:
I'd prefer to see us generate ExprTree's via parsing strings instead of making ExprTree nodes on the fly.  I do still harbor the fantasy of replacing the classad implementation someday by something other than a parse tree.